snapshots/devmapper/snapshotter_test.go

```diff
@@ -165,3 +140,85 @@ func TestMkfsXfsNonDefault(t *testing.T) {
 	err := mkfs(ctx, "xfs", "noquota", "")
 	assert.ErrorContains(t, err, `mkfs.xfs couldn't initialize ""`)
 }
+
+func TestMultipleXfsMounts(t *testing.T) {
+	testutil.RequiresRoot(t)
+
+	logrus.SetLevel(logrus.DebugLevel)
+
+	ctx := context.Background()
+	ctx = namespaces.WithNamespace(ctx, "testsuite")
+	tempDir, err := os.MkdirTemp("", "snapshot-suite-usage")
+	assert.NilError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	poolName := fmt.Sprintf("containerd-snapshotter-suite-pool-%d", time.Now().Nanosecond())
+	config := &Config{
+		RootPath:       tempDir,
+		PoolName:       poolName,
+		BaseImageSize:  "16Mb",
+		FileSystemType: "xfs",
+	}
+	snapshotter, closer, err := createSnapshotter(ctx, t, config)
+	assert.NilError(t, err)
+	defer closer()
+
+	var (
+		sizeBytes   int64 = 1048576 // 1MB
+		baseApplier       = fstest.Apply(fstest.CreateRandomFile("/a", 12345679, sizeBytes, 0777))
+	)
+
+	// Create base layer
+	mounts, err := snapshotter.Prepare(ctx, "prepare-1", "")
+	assert.NilError(t, err)
+
+	root1, _ := os.MkdirTemp(os.TempDir(), "containerd-mount")
+	defer func() {
+		mount.UnmountAll(root1, 0)
+		os.Remove(root1)
+	}()
+	err = mount.All(mounts, root1)
+	assert.NilError(t, err)
+	baseApplier.Apply(root1)
+	snapshotter.Commit(ctx, "layer-1", "prepare-1")
+
+	// Create one child layer
+	mounts, err = snapshotter.Prepare(ctx, "prepare-2", "layer-1")
+	assert.NilError(t, err)
+
+	root2, _ := os.MkdirTemp(os.TempDir(), "containerd-mount")
+	defer func() {
+		mount.UnmountAll(root2, 0)
+		os.Remove(root2)
+	}()
+	err = mount.All(mounts, root2)
+	assert.NilError(t, err)
+}
+
+func createSnapshotter(ctx context.Context, t *testing.T, config *Config) (snapshots.Snapshotter, func() error, error) {
+	// Create loopback devices for each test case
+	_, loopDataDevice := createLoopbackDevice(t, config.RootPath)
+	_, loopMetaDevice := createLoopbackDevice(t, config.RootPath)
+
+	err := dmsetup.CreatePool(config.PoolName, loopDataDevice, loopMetaDevice, 64*1024/dmsetup.SectorSize)
+	assert.NilError(t, err, "failed to create pool %q", config.PoolName)
+
+	snap, err := NewSnapshotter(context.Background(), config)
```

Comment:
How about taking the passed ctx instead of making a new one?

Reply:
Sounds good